unit-tests/py/rspy/usb_relay.py

```diff
@@ -0,0 +1,268 @@
+
+import re
+from rspy import log
+
+NO_RELAY = 0
+ACRONAME = 1
+YKUSH = 2
+
+
+class usb_relay:
+
+    def __init__(self):
+        self._relay = None
+        self._relay_type = NO_RELAY
+        self._is_connected = False
+
+    def __del__(self):
+        self.disable_ports()
+        self.disconnect()
+
+    def connect(self, reset = False):
+        if self.is_connected():
+            log.d("Already connected to", self._relay, "disconnecting...")
+            self.disconnect()
+        self._relay_type, self._relay = _find_active_relay()
+        if self._relay:
+            self._relay.connect(reset)
+            self._is_connected = True
+        log.d("Now connected to:", self._relay)
+
+    def disconnect(self):
+        if self._relay:
+            self._relay.disconnect()
+            del self._relay
+            self._relay = None
+            self._relay_type = NO_RELAY
+            self._is_connected = False
+
+    def is_connected(self):
+        return self._is_connected
+
+    def ports(self):
+        if self._relay_type != NO_RELAY:
+            return self._relay.ports()
+
+    def all_ports(self):
+        if self._relay_type != NO_RELAY:
+            return self._relay.all_ports()
+
+    def enable_ports(self, ports=None, disable_other_ports=False, sleep_on_change=0):
+        """
+        Set enable state to provided ports
+        :param ports: List of port numbers; if not provided, enable all ports
+        :param disable_other_ports: if True, the ports not in the list will be disabled
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.enable_ports(ports, disable_other_ports, sleep_on_change)
+        return False
+
+    def disable_ports(self, ports=None, sleep_on_change=0):
+        """
+        :param ports: List of port numbers; if not provided, disable all ports
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.disable_ports(ports, sleep_on_change)
+        return False
+
+    def recycle_ports(self, portlist=None, timeout=2):
+        """
+        Disable and enable a port
+        :param portlist: List of port numbers; if not provided, recycle all ports
+        :param timeout: how long to wait before re-enabling
+        :return: True if everything OK, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.recycle_ports(portlist=portlist, timeout=timeout)
+        return False
+
+    def find_all_hubs(self):
+        """
+        Yields all hub port numbers
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.find_all_hubs()
+
+    @property
+    def has_relay(self):
```

Comment:
Not sure I like this as a property. It certainly doesn't look like a member variable. I'd say just turn it into a function. Same for `not_supports_port_mapping`, etc.